ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -466,19 +470,48 @@ public Future<String> stitchBlob(BlobProperties blobProperties, byte[] userMetad
     }
     routerMetrics.operationQueuingRate.mark();
     FutureResult<String> futureResult = new FutureResult<>();
+    Callback<String> wrappedCallback =
+        restRequest != null && idConverter != null ? createIdConverterCallbackForStitch(restRequest, blobProperties,
+            futureResult, callback) : callback;
```

Comment:
there is one little thing about this callback and the future. It's possible that your callback would return exception but your future would return success. Let me explain. In getOperationController().stitchBlob method you passed in wrappedCallback and the futureResult, which means that when stitch blob succeeds, without the named blob operation, the futureResult will be set to success first (checkout completeOperation in this file), which means it will already succeeds, then the wrappedcallback is called. If the named blob operation failed, then the callback would return exception. In this case, the future and the callback returns different things.

Reply:
Adding a TODO.